### PR TITLE
Zoom scaling stroke

### PR DESCRIPTION
**Pull request by markmarkoh**
* Adds `vector-effect: non-scaling-stroke;` to `.datamap path, .datamap circle, .datamap line`
* Updates `defaults()` function to straight pass a function. Would be nice to find a cleaner way to copy a function.

### CODE REVIEW

File: dist/datamaps.all.js [hunk 1]
```diff
@@ -621,7 +621,7 @@
           // Deep copy if property not set
           if (obj[prop] == null) {
             if (typeof source[prop] == 'function') {
-              obj[prop] = source[prop].bind({});
+              obj[prop] = source[prop];
```
Comment (magic890) [inline dist/datamaps.all.js]: Why `.bind({})` is removed? It is useful to override `this` previously states.

Comment (markmarkoh) [inline dist/datamaps.all.js]: @magic890 calling `options.setProjection.apply(this, [options.element, options] );` on line 692 was getting ignored, `setProjection` was getting called but `this` was just `{}`.

File: src/examples/zoom-and-scale.html [hunk 2]
```diff
@@ -0,0 +1,233 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<body>
+  <script src="http://d3js.org/d3.v3.min.js"></script>
+  <script src="http://d3js.org/topojson.v1.min.js"></script>
+  <!-- I recommend you host this file on your own, since this will change without warning -->
+  <script src="/src/rel/datamaps.world.js?v=1"></script>
```
Comment (magic890) [inline src/examples/zoom-and-scale.html]: You are not using the compiled version of the library. So no `vector-effect: non-scaling-stroke`.

File: src/js/datamaps.js [hunk 3]
```diff
@@ -172,7 +172,7 @@
...
+      .html('.datamap path.datamaps-graticule { fill: none; stroke: #777; stroke-width: 0.5px; stroke-opacity: .5; pointer-events: none; } .datamap .labels {pointer-events: none;} .datamap path, .datamap circle, .datamap line {stroke: #FFFFFF; vector-effect: non-scaling-stroke; stroke-width: 1px;} .datamaps-legend dt, .datamaps-legend dd { float: left; margin: 0 3px 0 0;} .datamaps-legend dd {width: 20px; margin-right: 6px; border-radius: 3px;} .datamaps-legend {padding-bottom: 20px; z-index: 1001; position: absolute; left: 4px; font-size: 12px; font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;} .datamaps-hoverover {display: none; font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; } .hoverinfo {padding: 4px; border-radius: 1px; background-color: #FFF; box-shadow: 1px 1px 5px #CCC; font-size: 12px; border: 1px solid #CCC; } .hoverinfo hr {border:1px dotted #CCC; }');
```
Comment (magic890) [inline src/js/datamaps.js]: Applying `vector-effect: non-scaling-stroke` to all `.path` include also arcs.
This cause a weird behaviour on arcs using zoom.

Vector Effect on Arc

**Pull request merged**